Xamarin.PropertyEditing.Mac/Controls/CollectionInlineEditorControl.cs

```diff
@@ -25,8 +25,19 @@ public CollectionInlineEditorControl (IHostResourceProvider hostResources)
 			};
 
 			this.openCollection.Activated += (o, e) => {
-				CollectionEditorWindow.EditCollection (EffectiveAppearance, HostResources, ViewModel);
-				Window.MakeFirstResponder (this.openCollection);
+				NSWindow parentWindow = NSApplication.SharedApplication.KeyWindow;
```

Comment:
This may be a simpler implementation of this work around/hack. Then it doesn't need the additional helper function etc.

```suggestion
				var parentWindow = NSApplication.SharedApplication.KeyWindow;

				CollectionEditorWindow.EditCollection (EffectiveAppearance, HostResources, ViewModel);

				//HACK: after a modal display the GTK FocusedWindow, VS4M is null, we set the parent again so it behaves correctly
				//TODO: Remove this Hack, once VS4M is a proper Full Cream Cocoa app.
				parentWindow?.MakeKeyAndOrderFront (parentWindow);

				System.Threading.Tasks.Task.Delay (100).ContinueWith (t => {
					parentWindow?.MakeFirstResponder (this.openCollection);
				}, System.Threading.Tasks.TaskScheduler.FromCurrentSynchronizationContext ());
```